cider/tests/functionality/PrimitiveTypeArrowTest.cpp

```diff
@@ -0,0 +1,202 @@
+                                                  10,
+                                                  {"col_a", "col_b", "col_c"},
+                                                  {CREATE_SUBSTRAIT_TYPE(Bool),
+                                                   CREATE_SUBSTRAIT_TYPE(Fp32),
+                                                   CREATE_SUBSTRAIT_TYPE(I8)},
+                                                  {0, 0, 0},
+                                                  GeneratePattern::Random,
+                                                  0,
+                                                  0);
+  }
+};
+
+class PrimitiveTypeMixedTypeNullTest : public CiderTestBase {
+ public:
+  PrimitiveTypeMixedTypeNullTest() {
+    table_name_ = "test";
+    create_ddl_ = "CREATE TABLE test(col_a BOOLEAN, col_b DOUBLE, col_c INTEGER)";
+    QueryArrowDataGenerator::generateBatchByTypes(schema_,
+                                                  array_,
+                                                  10,
+                                                  {"col_a", "col_b", "col_c"},
+                                                  {CREATE_SUBSTRAIT_TYPE(Bool),
+                                                   CREATE_SUBSTRAIT_TYPE(Fp64),
+                                                   CREATE_SUBSTRAIT_TYPE(I32)},
+                                                  {1, 1, 1},
+                                                  GeneratePattern::Random,
+                                                  0,
+                                                  0);
+  }
+};
+
+class PrimitiveTypeMixedTypeHalfNullTest : public CiderTestBase {
+ public:
+  PrimitiveTypeMixedTypeHalfNullTest() {
+    table_name_ = "test";
+    create_ddl_ = "CREATE TABLE test(col_a BOOLEAN, col_b DOUBLE, col_c INTEGER)";
+    QueryArrowDataGenerator::generateBatchByTypes(schema_,
+                                                  array_,
+                                                  10,
+                                                  {"col_a", "col_b", "col_c"},
+                                                  {CREATE_SUBSTRAIT_TYPE(Bool),
+                                                   CREATE_SUBSTRAIT_TYPE(Fp64),
+                                                   CREATE_SUBSTRAIT_TYPE(I32)},
+                                                  {2, 2, 2},
+                                                  GeneratePattern::Random,
+                                                  0,
+                                                  0);
+  }
+};
+
+TEST_UNIT(PrimitiveTypeIntegerMaxTest, integerMaxTest)
+
+TEST_UNIT(PrimitiveTypeIntegerMINTest, integerMinTest)
+
+TEST_UNIT(PrimitiveTypeMixedTypeNotNullTest, mixedTypeNotNULLTest)
+
+TEST_UNIT(PrimitiveTypeMixedTypeNullTest, mixedTypeNULLTest)
+
+TEST_UNIT(PrimitiveTypeMixedTypeHalfNullTest, mixedTypeHalfNULLTest)
+
+int main(int argc, char** argv) {
+  testing::InitGoogleTest(&argc, argv);
+
```

Comment:
Add following lines at L195 for future easy debug:
`  logger::LogOptions log_options(argv[0]);
  log_options.parse_command_line(argc, argv);
  log_options.max_files_ = 0;  // stderr only by default
  logger::init(log_options);`